src/lib.rs

```diff
@@ -1,808 +1,268 @@
-//! `tsunami` provides an interface for running short-lived jobs and experiments on EC2 spot block
-//! instances. Most interaction with this library happens through
-//! [`TsunamiBuilder`](struct.TsunamiBuilder.html).
+//! `tsunami` provides an interface for running short-lived jobs and experiments on cloud
+//! instances.
 //!
-//! # Examples
+//! Most interaction with this library happens through
+//! [`TsunamiBuilder`](struct.TsunamiBuilder.html) and [`Tsunami`](struct.Tsunami.html).
+//!
+//! # Example
 //!
 //! ```rust,no_run
-//! # use tsunami::{Machine, MachineSetup, TsunamiBuilder};
-//! # use std::collections::HashMap;
-//! let mut b = TsunamiBuilder::default();
-//! b.use_term_logger();
-//! b.add_set(
-//!     "server",
-//!     1,
-//!     MachineSetup::new("m5.large", "ami-e18aa89b", |ssh| {
-//!         ssh.cmd("yum install nginx").map(|out| {
-//!             println!("{}", out);
-//!         })
-//!     }),
-//! );
-//! b.add_set(
-//!     "client",
-//!     3,
-//!     MachineSetup::new("m5.large", "ami-e18aa89b", |ssh| {
-//!         ssh.cmd("yum install wget").map(|out| {
-//!             println!("{}", out);
-//!         })
-//!     }),
-//! );
+//! use tsunami::TsunamiBuilder;
+//! use tsunami::providers::{Launcher, aws};
+//! use rusoto_core::DefaultCredentialsProvider;
+//! fn main() -> Result<(), failure::Error> {
```

Comment:
Like my comment on the `README` example, I'd like to see more features exercised and explained in comments here, as this is likely the primary place users are going to see "correct" tsunami code.

Reply:
The old main example is a good case-study:
https://github.com/jonhoo/tsunami/blob/96a2f912f445b57b02a217b4a6998837373989b6/src/lib.rs#L7-L38